animations-dbus/animations-dbus-server-surface.c

```diff
@@ -0,0 +1,691 @@
+                   effect_path,
+                   EFFECT_PATH_EXPECTED_COMPONENTS,
+                   path_length);
+      return FALSE;
+    }
+
+  if (!g_ascii_string_to_unsigned (effect_path_components[EFFECT_PATH_ANIMATION_MANAGER_INDEX],
+                                   10,
+                                   0,
+                                   G_MAXUINT,
+                                   &animation_manager_id64,
+                                   &local_error))
+    {
+      g_set_error (error,
+                   ANIMATIONS_DBUS_ERROR,
+                   ANIMATIONS_DBUS_ERROR_NO_SUCH_EFFECT,
+                   "Expected effect path component '%s' to be an "
+                   "unsigned integer: %s",
+                   effect_path_components[EFFECT_PATH_ANIMATION_MANAGER_INDEX],
+                   local_error->message);
+      return FALSE;
+    }
+
+  if (!g_ascii_string_to_unsigned (effect_path_components[EFFECT_PATH_ANIMATION_EFFECT_INDEX],
+                                   10,
+                                   0,
+                                   G_MAXUINT,
+                                   &animation_effect_id64,
+                                   &local_error))
+    {
+      g_set_error (error,
+                   ANIMATIONS_DBUS_ERROR,
+                   ANIMATIONS_DBUS_ERROR_NO_SUCH_EFFECT,
+                   "Expected effect path component '%s' to be an "
+                   "unsigned integer: %s",
+                   effect_path_components[EFFECT_PATH_ANIMATION_EFFECT_INDEX],
+                   local_error->message);
+      return FALSE;
+    }
+
+  *out_animation_manager_id = (unsigned int) animation_manager_id64;
+  *out_animation_effect_id = (unsigned int) animation_effect_id64;
+
+  return TRUE;
+}
+
+static gboolean
+animations_dbus_server_surface_attach_animation_effect (AnimationsDbusAnimatableSurface *animatable_surface,
+                                                        GDBusMethodInvocation           *invocation,
+                                                        const char                      *event,
+                                                        const char                      *effect_path)
+{
+  AnimationsDbusServerSurface *server_surface = ANIMATIONS_DBUS_SERVER_SURFACE (animatable_surface);
+  AnimationsDbusServerSurfacePrivate *priv =
+    animations_dbus_server_surface_get_instance_private (server_surface);
+  unsigned int animation_manager_id = 0;
+  unsigned int animation_effect_id = 0;
+  g_autoptr(GError) local_error = NULL;
+
+  /* Validate that the passed in effect_path is a valid object path */
+  if (!parse_effect_path (effect_path,
+                          &animation_manager_id,
+                          &animation_effect_id,
+                          &local_error))
+    {
+      g_dbus_method_invocation_return_gerror (invocation,
+                                              g_steal_pointer (&local_error));
+      return TRUE;
```

Comment:
Should the returns here be `TRUE` on failure?

Reply:
I should, yes - good catch.